crates/core/src/test_utils/deploy_helpers.rs

```diff
@@ -721,6 +723,77 @@ impl KakarotTestEnvironmentContext {
         }
     }
 
+    pub async fn from_dump_state() -> Self {
```

Comment:
we should probably add rust doc to this function, especially to list the different ways it can panic and with examples of panic cases